p2p/libp2p/identity/identity.go

```diff
@@ -0,0 +1,125 @@
+package identity
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+
+	"github.com/pokt-network/pocket/p2p/common"
+	"github.com/pokt-network/pocket/p2p/transport"
+	"github.com/pokt-network/pocket/p2p/types"
+	poktCrypto "github.com/pokt-network/pocket/shared/crypto"
+)
+
+var (
+	ErrIdentity = common.NewErrFactory("")
+)
+
+// PoktPeerFromStream builds a pokt peer from a libp2p stream.
+// (NOTE: excludes `ServiceURL`)
```

Comment:
1. What does this note mean?
2. When do you pick between NOTE vs NB?

Reply:
1. My thinking is that we have competing representations for peers, "pokt" and "libp2p". In order to conform to our current interfaces, we need a pokt peer (i.e. [`types.NetworkPeer`](https://github.com/pokt-network/pocket/blob/main/p2p/types/network_peer.go#L7)) in a context where we only have a libp2p stream to construct one from.
2. Apparently when I'm tired :sweat_smile: - `NB` is a habit I picked up for short comments, I don't have a strong opinion about it.